packages/nuxt/src/pages/runtime/plugins/router.ts

```diff
@@ -95,6 +96,8 @@ const plugin: Plugin<{ router: Router }> = defineNuxtPlugin({
     }
     nuxtApp.vueApp.use(router)
 
+    nuxtApp.vueApp.provide(RouterHashModeSymbol, hashMode)
```

Comment:
rather than using `provide`, I'd prefer to use a virtual module, if we can. it keeps it tree-shakeable and isn't dependent on the context.